cel/program.go

```diff
@@ -41,6 +42,9 @@ type Program interface {
 	// or `ProgramOption` values used in the creation of the evaluation environment or executable
 	// program.
 	Eval(vars interface{}) (ref.Val, *EvalDetails, error)
+
+	// Cost returns the estimated cost interval for the evaluation.
+	Cost() (min, max int64)
```

Comment:
Maybe make this a separate interface called `Coster`? You wouldn't need to embed the interface into other interfaces since the behavior of cost-ing is orthogonal to interpretation / attribute calculation. In theory the `Program`, `Interpretable` steps wouldn't be the only objects with a heuristic `Cost`, e.g. you could imagine adding it to a custom function implementation. 

If you make it a separate interface I'd wrap `Cost()` computations into a helper within the `interpreter` package:

```go
func estimateCost(i interface{}) (min, max int64) {
    c, ok := i.(Coster)
    if !ok {
       return 0, math.MaxInt64
    }
    return c.Cost()
}
```

Reply:
Note, if you created a separate interface, you could also expose a helper on the `cel` package: `cel.EstimateCost(Program) (min, max int64)`

Reply:
refactored. PTAL.

I created an issue https://github.com/google/cel-go/issues/371 but does not seem to have permission to link it here.